lib/fog/storage/openstack.rb

```diff
@@ -58,7 +58,15 @@ def self.reset
         def initialize(options = {})
           @openstack_api_key = options[:openstack_api_key]
           @openstack_username = options[:openstack_username]
-          @path = '/v1/AUTH_1234'
+          @openstack_management_url = options[:openstack_management_url] || 'http://example:8774/v2/AUTH_1234'
+
+          @openstack_management_uri = URI.parse(@openstack_management_url)
```

Comment:
This looks duplicated from `OpenStack::Core.authenticate`, which I guess is only included in the `Real` classes (e.g. in `storage/openstack.rb`). I looked around and couldn't find a better way to fit this. We could make a mixin that just parses the management URL, but it seems a bit silly. I think this is ok, but if you have an idea on how to avoid duplicating this code, I'd be all ears.